modules/pubsub/src/main/scala/com/snowplowanalytics/snowplow/sources/pubsub/PubsubSource.scala

```diff
@@ -186,8 +186,9 @@ object PubsubSource {
 
   private def runSubscriber[F[_]: Async](config: PubsubSourceConfig, control: Control): Resource[F, Unit] =
     for {
-      executor <- executorResource(Sync[F].delay(Executors.newScheduledThreadPool(2 * config.parallelPullCount)))
       direct <- executorResource(Sync[F].delay(MoreExecutors.newDirectExecutorService()))
+      parallelPullCount = chooseNumParallelPulls(config)
+      executor <- executorResource(Sync[F].delay(Executors.newScheduledThreadPool(2 * parallelPullCount)))
```

Comment:
This `2` already has been here before, but could you explain to me like I'm five - why 2? 😁 And does new config affect this `2` and should be adjusted as well?

Reply:
If we don't supply our own executor, then [here is where](https://github.com/googleapis/java-pubsub/blob/v1.129.5/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java#L699) the google sdk picks the number of threads.

```java
int threadCount = Math.max(6, 2 * parallelPullCount);
```

So previously I copied that.  I think it is `2 *` because the thread pool is used for several tasks, e.g. processing events, acking, extending ack deadlines (sorry I can't remember all the details exactly).

I do not think this needs adjusting for the new change.

You might ask if we should limit it to max 6 threads, same as the sdk.  I don't think so.  Because of how we block on threads in some of our snowplow apps, I think there is no harm in exceeding 6 slightly.